tests/unit/python/foglamp/services/core/api/test_notification.py

```diff
@@ -0,0 +1,272 @@
+from foglamp.common.storage_client.storage_client import StorageClientAsync
+from foglamp.common.service_record import ServiceRecord
+from foglamp.common.configuration_manager import ConfigurationManager
+from foglamp.services.core.service_registry.service_registry import ServiceRegistry
+from foglamp.services.core.service_registry import exceptions as service_registry_exceptions
+from foglamp.common.audit_logger import AuditLogger
+from foglamp.services.core.api import notification
+
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+mock_registry = [ServiceRecord(uuid.uuid4(), "Notifications", "Notification", "http", "localhost", "8118", "8118")]
+rule_config = {
+    "threshold": {
+        "plugin": {
+            "description": "The accepted tolerance",
+            "default": "threshold",
+            "value": "threshold",
+            "type": "string"
+        },
+        "tolerance": {
+            "description": "The accepted tolerance",
+            "default": "4",
+            "value": "4",
+            "type": "integer"
+        },
+        "window": {
+            "description": "The window to perform rule evaluation over in minutes",
+            "default": "60",
+            "value": "60",
+            "type": "integer"
+        },
+        "trigger": {
+            "description": "Temparature threshold value",
+            "default": "40",
+            "value": "40",
+            "type": "integer"
+        },
+        "asset": {
+            "description": "The asset the notification is defined against",
+            "default": "temperature",
+            "value": "temperature",
+            "type": "string"
+        }
+    }
+}
+
+delivery_config = {"email": {
+        "plugin": {
+            "description": "Email",
+            "default": "email",
+            "value": "email",
+            "type": "string"
+        },
+        "server": {
+            "description": "The smtp server",
+            "default": "smtp",
+            "value": "smtp",
+            "type": "string"
+        },
+        "from": {
+            "description": "The from address to use in the email",
+            "default": "foglamp",
+            "value": "foglamp",
+            "type": "string"
+        },
+        "to": {
+            "description": "The address to send the notification to",
+            "default": "test",
+            "value": "test",
+            "type": "string"
+        }
+    },
+    "sms": {
+        "plugin": {
+            "description": "SMS",
+            "default": "sms",
+            "value": "sms",
+            "type": "string"
+        },
+        "number": {
+            "description": "The phone number to call",
+            "default": "07812 343830",
+            "value": "07812 343830",
+            "type": "string",
+        }
+    }
+}
+
+NOTIFICATION_TYPE = ["one shot", "retriggered", "toggled"]
```

Comment:
better to import this symbol from the real code and, also assert this in `test_supported_notification_type()`